changelog.d/pr-10014

```diff
@@ -0,0 +1,11 @@
+synopsis: Update ghc args normalization and ghc option rendering
+packages: Cabal
+issues: #9729
+prs: #10014
+
+description: {
+
+The flags fdiagnostics-as-json, fprint-error-index-lists, fbreak-points, dipe-stats, ffamily-application-cache, fprint-redundant-promotion-ticks, show-error-context and unoptimized-core-for-interpreter have been added to the flags that do not cause recompilation. 
```

Comment:
Please, make sure that all flag names are spelled correctly. E.g. they should have `-f` in front of them if that's what GHC wants. Both here (in the changelog), and in the PR description. geekosaur pointed out some issues in the PR description.